src/viam/examples/modules/example_mlmodel.cpp

```diff
@@ -0,0 +1,240 @@
+        static constexpr std::array<float, 1> category_data = {0};
+        static constexpr std::array<float, 1> score_data = {.99};
+        static constexpr std::array<float, 1> num_dets_data = {1};
+
+        auto location_tensor =
+            make_tensor_view(location_data.data(), location_data.size(), {1, 1, 4});
+
+        auto category_tensor = make_tensor_view(category_data.data(), category_data.size(), {1, 1});
+
+        auto score_tensor = make_tensor_view(score_data.data(), score_data.size(), {1, 1});
+
+        auto num_dets_tensor = make_tensor_view(num_dets_data.data(), num_dets_data.size(), {1});
+
+        using namespace std::literals::string_literals;
+        named_tensor_views tensors{{"location"s, std::move(location_tensor)},
+                                   {"category"s, std::move(category_tensor)},
+                                   {"score"s, std::move(score_tensor)},
+                                   {"n_detections"s, std::move(num_dets_tensor)}};
+
+        return std::make_shared<named_tensor_views>(std::move(tensors));
+    }
+
+    struct metadata metadata() override {
+        std::cout << "ExampleMLModelService: recieved `metadata` invocation" << std::endl;
+
+        // This metadata is modelled on the results obtained from
+        // invoking `Metadata` on a instance of tflite_cpu configured
+        // per the instructions and data at
+        // https://github.com/viamrobotics/vision-service-examples/tree/aa4195485754151fccbfd61fbe8bed63db7f300f
+
+        return {// `name`
+                "C++ SDK Example MLModel - Faking EfficientDet Lite0 V1",
+
+                // `type`
+                "fake_tflite_detector",
+
+                // `description`
+                "Identify which of a known set of objects might be present and provide "
+                "information "
+                "about their positions within the given image or a video stream.",
+
+                // `inputs`
+                {
+                    {// `name`
+                     "image",
+
+                     // `description`
+                     "Input image to be detected.",
+
+                     // `data_type`
+                     tensor_info::data_types::k_uint8,
+
+                     // `shape`
+                     {1, 320, 320, 3},
+
+                     // `associated_files`
+                     {},
+
+                     // `extra`
+                     {}},
+                },
+
+                // `outputs`
+                {{// `name`
+                  "location",
+
+                  // `description`
+                  "The locations of the detected boxes.",
+
+                  // `data_type`
+                  tensor_info::data_types::k_float32,
+
+                  // `shape`
+                  //
+                  // TODO(RSDK-3033): Once tflite_cpu reports output
+                  // shape metadata, fill this in.
+                  {},
```

Comment:
can't you fill this in anyway, since the API has the field available (even if the model can't do anything with it yet)?

Reply:
@bhaney - Can you elaborate a bit on what you are proposing? I'm not quite getting it.